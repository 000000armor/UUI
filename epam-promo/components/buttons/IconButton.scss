@use  '../../assets/styles/index' as *;

.root {
    fill: var(--uui-icon-btn-color);

    &:global(.-clickable) {
        &:hover, &:focus {
            fill: var(--uui-icon-btn-color-hover);
        }

        &:active {
            fill: var(--uui-icon-btn-color-active);
        }
    }

    &:global(.uui-disabled) {
        fill: var(--uui-icon-btn-color-disabled);

        &:global(.-clickable) {
            &:hover, &:focus {
                fill: var(--uui-icon-btn-color-disabled);
            }

            &:active {
                fill: var(--uui-icon-btn-color-disabled);
            }
        }
    }
}
