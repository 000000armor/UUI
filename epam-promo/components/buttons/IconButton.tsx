import React from 'react';
import { withMods } from '@epam/uui-core';
import { allEpamPrimaryColors, EpamPrimaryColor } from '../types';
import { IconButton as uuiIconButton, IconButtonProps as UuiIconButtonProps } from '@epam/uui';


export type IconColor = EpamPrimaryColor | 'gray30' | 'gray50' | 'gray60';
export const allIconColors: IconColor[] = [...allEpamPrimaryColors, 'gray30', 'gray50', 'gray60'];

export interface IconButtonMods {
    color?: IconColor;
}

function applyIconButtonMods(mods: Omit<UuiIconButtonProps, 'color'> & IconButtonMods) {
    return [
        'uui-theme-promo',
        [`icon-button-color-${ mods.color || "gray60" }`],
    ];
}

export type IconButtonProps = Omit<UuiIconButtonProps, 'color'> & IconButtonMods;

export const IconButton = withMods<Omit<UuiIconButtonProps, 'color'>, IconButtonMods>(uuiIconButton, applyIconButtonMods);
