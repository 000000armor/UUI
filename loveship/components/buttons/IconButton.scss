@use '../../assets/styles/scss/index'  as *;

.root {
    overflow: visible;
    fill: var(--uui-color-700);

    &:global(.-clickable:not(.uui-disabled)) {
        &:hover, &:focus {
            fill: var(--uui-color-800);
        }

        &:active {
            fill: var(--uui-color-900);
        }
    }

    &:global(.uui-disabled) {
        fill: $night500;

        &:hover {
            fill: $night500;
        }
    }

    &:global(.uui-invalid) {
        fill: $fire;
    }
}
