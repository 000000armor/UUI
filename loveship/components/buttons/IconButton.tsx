import { IconButton as uuiIconButton, IconButtonProps as UuiIconButtonProps } from '@epam/uui';
import { devLogger, withMods } from '@epam/uui-core';
import { commonControlColors } from '../types';

export type IconColor = (typeof commonControlColors)[number];
export const allIconColors: IconColor[] = commonControlColors;
export const deprecatedIconButtonColors = ['night200', 'night300', 'night400'];

export interface IconButtonMods {
    color?: IconColor;
}

export type IconButtonProps = Omit<UuiIconButtonProps, 'color'> & IconButtonMods;

export const IconButton = withMods<Omit<UuiIconButtonProps, 'color'>, IconButtonMods>(
    uuiIconButton,
    () => [],
    (props) => {
        devLogger.warnAboutDeprecatedPropValue<IconButtonProps, 'color'>({
            component: 'IconButton',
            propName: 'color',
            propValue: props.color,
            condition: () => deprecatedIconButtonColors.indexOf(props.color) !== -1,
        });
        return {
            color: props.color ?? 'night600',
        } as IconButtonProps;
    },
);
