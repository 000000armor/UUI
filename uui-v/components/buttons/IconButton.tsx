import { IconButtonBaseProps, IconButton as uuiIconButton } from '@epam/uui-components';
import { withMods } from '@epam/uui';
import * as css from './IconButton.scss';
import * as styles from '../../assets/styles/colorvars/buttons/iconButton-colorvars.scss';

export type IconColor = 'info' | 'success' | 'warning' | 'error' | 'secondary' | 'default';
export const allIconColors: IconColor[] = ['info', 'success', 'warning', 'error', 'secondary', 'default'];

export interface IconButtonMods {
    color?: IconColor;
}

export interface IconButtonProps extends IconButtonBaseProps, IconButtonMods { }

function applyIconButtonMods(mods: IconButtonProps & IconButtonMods) {
    return [
        css.root,
        styles['icon-color-' + (mods.color || 'default')],
    ];
}

export const IconButton = withMods<IconButtonProps, IconButtonMods>(uuiIconButton, applyIconButtonMods);