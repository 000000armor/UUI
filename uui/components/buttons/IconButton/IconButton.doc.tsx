import * as React from 'react';
import { allIconColors, IconButton, IconButtonProps } from './IconButton';
import { ColorPicker, DocBuilder } from '@epam/uui-docs';
import { onClickDoc, isDisabledDoc, iCanRedirectDoc, isInvalidDoc } from '../../../docs';
import { DefaultContext } from '../../../docs';

const IconButtonDoc = new DocBuilder<IconButtonProps>({ name: 'IconButton', component: IconButton })
    .implements([onClickDoc, isDisabledDoc, isInvalidDoc, iCanRedirectDoc])
    .prop('color', { renderEditor: (editable: any, examples) => <ColorPicker colors={ examples.map(i => ({ value: i })) } { ...editable } />, examples: allIconColors })
    .withContexts(DefaultContext);

export default IconButtonDoc;
