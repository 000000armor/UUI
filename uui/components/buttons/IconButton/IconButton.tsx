import { IconButtonBaseProps, IconButton as uuiIconButton } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import * as css from './IconButton.scss';
import './IconButton.vars.scss';
import { uuiComponentClass } from '../../constant';

export type IconColor = 'info' | 'success' | 'warning' | 'error' | 'secondary' | 'default';
export const allIconColors: IconColor[] = ['info', 'success', 'warning', 'error', 'secondary', 'default'];

export interface IconButtonMods {
    color?: IconColor;
}

export interface IconButtonProps extends IconButtonBaseProps, IconButtonMods { }

function applyIconButtonMods(mods: IconButtonProps & IconButtonMods) {
    return [
        uuiComponentClass.iconButton,
        `icon-button-color-${mods.color || 'default'}`,
        css.root,
    ];
}

export const IconButton = withMods<IconButtonProps, IconButtonMods>(uuiIconButton, applyIconButtonMods);