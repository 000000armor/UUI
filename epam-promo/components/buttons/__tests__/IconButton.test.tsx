import React from 'react';
import { IconButton } from '../IconButton';
import { SvgMock, renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('IconButton', () => {
    it('should render with minimum props', async () => {
        const tree = await renderSnapshotWithContextAsync(<IconButton />);
        expect(tree).toMatchSnapshot();
    });

    it('should render with maximum props', async () => {
        const tree = await renderSnapshotWithContextAsync(<IconButton color="blue" onClick={ jest.fn } icon={ SvgMock } isDisabled={ false } />);
        expect(tree).toMatchSnapshot();
    });
});
