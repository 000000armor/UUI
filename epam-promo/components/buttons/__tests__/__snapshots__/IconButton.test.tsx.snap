// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should be rendered correctly 1`] = `
<button
  className="root icon-color-gray60 uui-button-box uui-enabled"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={null}
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  />
</button>
`;

exports[`IconButton should be rendered correctly with props 1`] = `
<button
  className="root icon-color-blue uui-button-box uui-enabled -clickable"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#calendar-18.svg"
      />
    </svg>
  </div>
</button>
`;
